pyproject.toml

```diff
@@ -44,7 +44,7 @@ PyGithub="2.4.0"
 pyperclip="1.9.0"
 ruamel-yaml = "^0.18.6"
 taskcluster = "^56.0.3"
-taskcluster-taskgraph = "^14.1.1"
+taskcluster-taskgraph = "^14.2.1"
```

Comment:
I noticed we have version `v14.2.1` pinned in `.taskcluster.yml` but here we use `^`, so it can be automatically bumped on the next `poetry lock`. Is it expected?

Reply:
That's a good callout; ideally we'd keep those in sync. The one in `.taskcluster.yml` is used when taskgraph is run during decision tasks. As I understand it, the one here is used for local tests and the `taskgraph` tests that run on pushes/prs.

I'll send a follow-up once this lands to pin to the specific version in `pyproject.toml`, and add comments everywhere about keeping it in sync.

Reply:
https://github.com/mozilla/translations/pull/1124 for this.